Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,434 @@
+
+        public int DataProcessing
+        {
+            get => Parse(CurrentPerson.DataProcessing);
+            set
+            {
+                if (CurrentPerson.DataProcessing != value.ToString())
+                {
+                    CurrentPerson.DataProcessing = value.ToString();
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public int DataProcessingMod
+        {
+            get => Parse(CurrentPerson.ModDataProcessing) + (CurrentPerson.Overclocked == "DataProcessing" ? 1 : 0);
+            set
+            {
+                value -= (CurrentPerson.Overclocked == "DataProcessing" ? 1 : 0);
+                if (string.IsNullOrEmpty(CurrentPerson.ModDataProcessing))
+                    CurrentPerson.ModDataProcessing = "0";
+                if (CurrentPerson.ModDataProcessing != value.ToString())
+                {
+                    CurrentPerson.ModDataProcessing = value.ToString();
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public int TotalDataProcessing => Parse(CurrentPerson.DataProcessing) + Parse(CurrentPerson.ModDataProcessing) + (CurrentPerson.Overclocked == "DataProcessing" ? 1 : 0);
+
+        public int Firewall
+        {
+            get => Parse(CurrentPerson.Firewall);
+            set
+            {
+                if (CurrentPerson.Firewall != value.ToString())
+                {
+                    CurrentPerson.Firewall = value.ToString();
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public int FirewallMod
+        {
+            get => Parse(CurrentPerson.ModFirewall) + (CurrentPerson.Overclocked == "Firewall" ? 1 : 0);
+            set
+            {
+                value -= (CurrentPerson.Overclocked == "Firewall" ? 1 : 0);
+                if (string.IsNullOrEmpty(CurrentPerson.ModFirewall))
+                    CurrentPerson.ModFirewall = "0";
+                if (CurrentPerson.ModFirewall != value.ToString())
+                {
+                    CurrentPerson.ModFirewall = value.ToString();
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public int TotalFirewall => Parse(CurrentPerson.Firewall) + Parse(CurrentPerson.ModFirewall) + (CurrentPerson.Overclocked == "Firewall" ? 1 : 0);
+
+        public int ActionDicePool => GetTotalAttribute(CurrentAction.ActionAttribute) +
+                GetTotalSkill(CurrentAction.ActionSkill) +
+                CurrentAction.ActionModifier +
+                WoundModifier +
+                ActionModifier;
+
+        public int DefenceDicePool => GetTotalSkill(CurrentAction.DefenceSkill) +
+                GetTotalAttribute(CurrentAction.DefenceAttribute) +
+                WoundModifier +
+                CurrentAction.DefenceModifier;
```

Comment:
I have touched on this in another comment, but your calculated pools for a particular action need to be able to support more advanced patterns and arithmetic. This setup works for a specific set of cases and will absolutely misbehave outside of that specific set. And in some cases, that misbehavior also covers RAW behavior, c.f. my comments about Full Matrix Defense and I Am the Firewall.